chevah/python-package
# [#3418] Build for Ubuntu 16.04 and update Linux, OpenBSD and OSX builds to cffi.

Scope
=====

Update build system to create package for ubuntu 16.04

Changes
========

Since pyopenssl 0.13 no longer works with OpenSSL available on Ubuntu 16.04  i have updated the build system to the one similar to Arch.

In the future all builds should use that system.

I have refactored a bit the windows part to re-use the common process.

src/python/chevahbs no longer installs pip, but just make sure that we have a minimal python.

I have also enable new build system for rhel 5,6,7 and sles 11,12 and generic linux, raspbian7, openbsd and solaris11

aix, freebsd and solaris10 are broken ... but since solaris 10 has a very old openssl I am not sure if we can upgrade it.

As a drive by I have added the github PR template.

How to test
=========

reviewers: @brunogola @dumol @alibotean 

check that changes make sense

thanks!
